cmd/snap-confine/mount-support-nvidia.c

```diff
@@ -201,7 +201,7 @@ static void sc_populate_libgl_with_hostfs_symlinks(const char *libgl_dir,
 			hostfs_symlink_target[0] = 0;
 			num_read =
 			    readlink(pathname, hostfs_symlink_target,
-				     sizeof hostfs_symlink_target);
+				     sizeof hostfs_symlink_target - 1);
```

Comment:
This is an off-by-one bug introduced in 740eba60374cfad24eccef1e35019be6a3d4b5f2 which was before snap-confine was pulled into the snapd project. The PR was https://github.com/snapcore/snap-confine/pull/36 (looks like it had no reviews?!?, though I may be misremembering how the reviews were done back then).

Regardless, the find is legitimate because while we `hostfs_symlink_target[num_read] = 0;` 4 lines later, this scribbles past the end of the buffer when the symlink target is exactly 512 long, which is wrong. We are protected by the use of sc_must_snprintf() when using this to populate symlink_target so this happens to not be a bug that can bite. Even if it were, it would not be security relevant because it would require root to subvert, so no privilege escalation.

In addition to this change, please use: `char hostfs_symlink_target[512] = { 0 };` in the initialization, up above.

Reply:
Back then we didn't do reviews the way we do now.

Reply:
Initialisation done.